src/main/java/com/exasol/cloudwatch/CloudWatchAdapter.java

```diff
@@ -78,11 +78,10 @@ private void runSynchronization(final Instant minuteToReport, final Connection e
 
     private Connection connectToExasol() {
         try {
-            return DriverManager
-                    .getConnection(
-                            "jdbc:exa:" + this.configuration.getExasolHost() + ":" + this.configuration.getExasolPort()
-                                    + ";schema=SYS",
-                            this.configuration.getExasolUser(), this.configuration.getExasolPass());
+            final ExasolCredentials exasolCredentials = this.configuration.getExasolCredentials();
+            return DriverManager.getConnection(
+                    "jdbc:exa:" + exasolCredentials.getHost() + ":" + exasolCredentials.getPort() + ";schema=SYS",
+                    exasolCredentials.getUser(), exasolCredentials.getPass());
```

Comment:
Do we close this connection afterwards?

You can also add a finally close here, so that it is closed between the lambda invocations.

Reply:
yes. It's in a try-with-resources in line 56